kommunicate/src/main/java/io/kommunicate/async/KmConversationCreateTask.java

```diff
@@ -0,0 +1,73 @@
+package io.kommunicate.async;
+
+import android.content.Context;
+import android.os.AsyncTask;
+
+import com.applozic.mobicomkit.api.people.ChannelInfo;
+import com.applozic.mobicomkit.channel.service.ChannelService;
+import com.applozic.mobicomkit.feed.ChannelFeedApiResponse;
+
+import java.lang.ref.WeakReference;
+
+import io.kommunicate.callbacks.KMStartChatHandler;
+import io.kommunicate.callbacks.KmStartConversationHandler;
+
+public class KmConversationCreateTask extends AsyncTask<Void, Void, ChannelFeedApiResponse> {
+    private WeakReference<Context> context;
+    private ChannelService channelService;
+    private ChannelInfo channelInfo;
+    private KmStartConversationHandler startConversationHandler;
+    private KMStartChatHandler startChatHandler;
+
+
+    public KmConversationCreateTask(Context context, ChannelInfo channelInfo, KmStartConversationHandler startConversationHandler) {
+        this.context = new WeakReference<>(context);
```

Comment:
We can create another constructor with common elements and then call this(..,..,..) from these 2

Reply:
Done. Actually, this was done as the second constructor is deprecated and will be removed after some time.